ci/ubuntu20.04-gcc11-x64.dockerfile

```diff
@@ -59,11 +59,14 @@ RUN if [ "$NimbleSM_ENABLE_MPI" = "ON" ]; then \
     else \
         mv /opt/spack-environment/spack-serial.yaml /opt/spack-environment/spack.yaml && rm /opt/spack-environment/spack-mpi.yaml; \
     fi
+# create pre_nimble environment from spack.yaml and concretize
 RUN cd /opt/spack-environment \
-  && . /opt/spack/share/spack/setup-env.sh \
-  && spack env activate . \
-  && spack install --fail-fast \
-  && spack gc -y
+  && . /opt/spack/share/spack/setup-env.sh && spack env create pre_nimble /opt/spack-environment/spack.yaml\
+  && spack env activate pre_nimble && spack concretize && spack env deactivate
+# make nimble env from lock
+RUN . /opt/spack/share/spack/setup-env.sh && spack env create nimble /opt/spack/var/spack/environments/pre_nimble/spack.lock
+# activate nimble env and install
+RUN . /opt/spack/share/spack/setup-env.sh && spack env activate nimble && spack install --fail-fast && spack gc -y
```

Comment:
That's right. The idea is to install in an environment created from a spack.lock file (concrete specifications) and not spack.yaml(abstract specifications that could change or be met in multiple ways). The spack.lock file is generated by concretizing the first environment. The environment we install with(from the spack.lock file) will then always be the same so that layer can be cached. This avoids the rebuilding of all the spack dependencies.